catboost/python-package/catboost/core.py

```diff
@@ -1476,6 +1492,13 @@ def set_feature_names(self, feature_names):
         '''
         self._object._set_feature_names(feature_names)
 
+
+    def _get_tags(self):
+        tags = _CATBOOST_SKLEARN_COMPAT_TAGS
+        if self._init_params['task_type'] == 'GPU':
```

Comment:
_init_params might not have task_type. Also please instead of having default value and changing it for GPU, set it dependand on task_type